src/package_report.py

```diff
@@ -221,6 +221,35 @@ def _generate_python_package_size_report_per_image(
     return validate_result
 
 
+def _generate_python_package_dependency_report(image_config, base_version_dir, target_version_dir):
+    env_in_file_name = image_config["build_args"]["ENV_IN_FILENAME"]
+
+    base_match_spec_out = get_match_specs(base_version_dir + "/" + env_in_file_name) if base_version_dir else dict()
+    target_match_spec_out = get_match_specs(target_version_dir + "/" + env_in_file_name)
+
+    base_packages_match_spec_out = {k: v for k, v in base_match_spec_out.items()}
+    target_packages_match_spec_out = {k: v for k, v in target_match_spec_out.items()}
+
+    results = dict()
+    for package, target_match_spec_out in target_packages_match_spec_out.items():
+        # We only care about the newly introduced marquee packages
+        if package not in base_packages_match_spec_out:
+            # Pull package metadata from conda-forge and dump into json file
```

Comment:
For this package, you should actually query env.out to get the corresponding Match spec (which includes the version that is used during the build.)

Then we should use the env.out match spec in the search query